_posts/2019-12-16-event-store-6.0.0-preview-release.md

```diff
@@ -0,0 +1,386 @@
+
+```c#
+var eventStoreClient = new EventStore.Grpc.EventStoreGrpcClient(new Uri("https://localhost:2113/"),
+    () => new HttpClient(new SocketsHttpHandler
+    {
+        SslOptions = new SslClientAuthenticationOptions
+        {
+            RemoteCertificateValidationCallback = delegate { return true; }
+        }
+    }));
+```
+
+### Writing to a Stream with Expected Version Any
+
+```c#
+var eventData = new EventData(Uuid.NewUuid(), "type", JsonSerializer.SerializeToUtf8Bytes(new SomeEvent()));
+await eventStoreClient.AppendToStreamAsync(
+  "stream", AnyStreamRevision.Any, new[] { eventData }, cancellationToken: cancellationToken);
+```
+
+### Writing to a Stream with Expected Version
+
+```c#
+var eventData = new EventData(Uuid.NewUuid(), "type", JsonSerializer.SerializeToUtf8Bytes(new SomeEvent()));
+await eventStoreClient.AppendToStreamAsync(
+  "stream", new StreamRevision(5), new[] { eventData }, cancellationToken: cancellationToken);
+```
+
+### Reading a Stream
+
+```c#
+var events = eventStoreClient.ReadStreamForwardsAsync(
+  streamName, StreamRevision.Start, int.MaxValue, cancellationToken: cancellationToken);
+```
+
+### Subscribe to All
+
+```c#
+using var subscription = eventStoreClient.SubscribeToAll(
+  async (s, e, ct) => { await …; }, userCredentials: new UserCredentials(“admin”, “changeit”));
+```
+
+### Subscribe to a Stream
+
+```c#
+using var subscription = eventStoreClient.SubscribeToStream(
+  streamName, async (s, e, ct) => await ...);
+```
+
+## Server Side Filtering
+
+We have also added the ability to filter both reads and subscriptions on the server-side. You can filter by either event type or stream name.
+
+The intention here is to remove the need for having to setup and use projections when you only want to subscribe to a stream containing specific events.
+
+New methods and overloads have been added to the .NET client API for reads and subscriptions.  
+New HTTP endpoints have been added as well for reading only.
+
+The API differs depending on whether you use the TCP or gRPC client. Below are some examples to help you get started.
+
+### TCP Client
+
+The server side filter is created using the new Filter class and you can choose to match either the event type or stream name against a prefix or regular expression.
+
+```c#
+var filter = Filter.EventType.Prefix("SimpleEvent");
+```
+
+That filter can then be used when making calls to either `FilteredReadAllEventsForwardAsync` or `FilteredReadAllEventsBackwardAsync` on the connection.
+
+```c#
+var readEvents = await connection.ReadAllEventsForwardFilteredAsync(
+    position: Position.Start, 
+    maxCount: 4096, 
+    resolveLinkTos: false,
+    eventFilter: eventFilter,
+    maxSearchWindow: 1000);
+```
+
+There is also an additional overload where you can provide a max search window argument.  
+This specifies the maximum number of events the server will search through before returning the slice.
```

Comment:
If no events are found within the window, an empty slice will be returned... @mat-mcloughlin ?